setup.py

```diff
@@ -76,8 +93,32 @@
         include_package_data=True,
         extras_require={
             'alldeps': (
-                'numpy >= {}'.format(NUMPY_MIN_VERSION),
-                'scipy >= {}'.format(SCIPY_MIN_VERSION),
+                f'numpy>={NUMPY_MIN_VERSION}',
+                f'scipy>={SCIPY_MIN_VERSION}',
+                f'joblib>={JOBLIB_MIN_VERSION}',
+                f'threadpoolctl>={THREADPOOLCTL_MIN_VERSION}'
+            ),
+            'examples': (
+                f'matplotlib>={MATPLOTLIB_MIN_VERSION}',
+                f'scikit-image>={SCIKIT_IMAGE_MIN_VERSION}',
+                f'pandas>={PANDAS_MIN_VERSION}',
+                f'seaborn>={SEABORN_MIN_VERSION}',
+                f'Pillow>={PILLOW_MIN_VERSION}',
+                f'memory_profiler>={MEMORY_PROFILER_MIN_VERSION}'
+            ),
+            'tests': (
+                f'pytest>={PYTEST_MIN_VERSION}',
+                f'pytest-cov>={PYTEST_COV_MIN_VERSION}',
+                f'Cython>={CYTHON_MIN_VERSION}',
+                f'flake8>={FLAKE8_MIN_VERSION}',
+                f'mypy>={MYPY_MIN_VERSION}',
+                f'coverage>={COVERAGE_MIN_VERSION}'
+            ),
+            'docs': (
+                f'sphinx>={SPHINX_MIN_VERSION}',
+                f'sphinx-gallery>={SPHINX_GALLERY_MIN_VERSION}',
+                f'packaging>={PACKAGING_MIN_VERSION}',
```

Comment:
packaging is vendored in setuptools since 7.0 do we really need it? The contributing instructions may be inaccurate.

Reply:
OK, it is mentioned in the build script for the doc:
https://github.com/scikit-learn/scikit-learn/blob/e9340e3f4b1d7c285dc02062cd594ab457a5c268/build_tools/circle/build_doc.sh#L166-L172

and it contributing guidelines
https://github.com/scikit-learn/scikit-learn/blob/e1c44ba5037e22ac2c59d104f3fd96866fc9f76d/doc/developers/contributing.rst#L585

Should I remove those comments too?

Reply:
No actually looks like it's still needed because we need a more recent version than the one shiped in setuptools https://github.com/scikit-learn/scikit-learn/pull/15566#discussion_r344143887